storage/rocksdb/ha_rocksdb.cc

```diff
@@ -5060,26 +5171,35 @@ class Rdb_transaction {
 
   int rollback_to_savepoint(void *const savepoint MY_ATTRIBUTE((__unused__))) {
     if (has_modifications()) {
+      assert(statement_snapshot_type == snapshot_type::CURRENT ||
+             statement_snapshot_type == snapshot_type::EXPLICIT);
+
       my_error(ER_ROLLBACK_TO_SAVEPOINT, MYF(0));
       m_rollback_only = true;
       return HA_EXIT_FAILURE;
     }
     return HA_EXIT_SUCCESS;
   }
 
-  /*
-    This is used by transactions started with "START TRANSACTION WITH "
-    "CONSISTENT [ROCKSDB] SNAPSHOT". When tx_read_only is turned on,
-    snapshot has to be created via DB::GetSnapshot(), not via Transaction
-    API.
-  */
-  bool is_tx_read_only() const { return m_tx_read_only; }
+  [[nodiscard]] bool is_tx_read_only() const noexcept {
```

Comment:
how about add _snapshot into method name?

Reply:
It is the whole transaction that is read only